ext/sqlite3/database.c

```diff
@@ -12,6 +12,39 @@
 
 VALUE cSqlite3Database;
 
+static void
+close_or_discard_db(sqlite3RubyPtr ctx)
+{
+    if (ctx->db) {
+        if (ctx->owner == getpid()) {
+            // Ordinary close.
+            sqlite3_close_v2(ctx->db);
+        } else {
+            // This is an open connection carried across a fork().
+            // "Discard" it. See adr/2024-09-fork-safety.md
+            sqlite3_file *sfile;
+            int status;
+
+            rb_warning("An open sqlite database connection was inherited from a forked process and "
+                       "is being discarded. This is a memory leak. If possible, please close all sqlite "
+                       "database connections before forking.");
+
+            // close the open file descriptors
+            status = sqlite3_file_control(ctx->db, NULL, SQLITE_FCNTL_FILE_POINTER, &sfile);
```

Comment:
`dbName` is always `NULL`? I can't find in the API in which case you have other databases open on a single connection.

Reply:
From https://sqlite.org/c3ref/file_control.html :

> The name of the database is "main" for the main database or "temp" for the TEMP database, or the name that appears after the AS keyword for databases that are added using the [ATTACH](https://sqlite.org/lang_attach.html) SQL command. A NULL pointer can be used in place of "main" to refer to the main database file.

Reply:
Right so potentially you'd need to close more than just the main database?

Reply:
Ah, I misunderstood what you were saying. I'll poke around.

Reply:
Relevant section of the docs is https://www.sqlite.org/inmemorydb.html

Reply:
Just pushed a commit that should make sure we get all the named databases.